File: Implementation/javascript/16967.js
const input = require("fs").readFileSync("/dev/stdin").toString().split("\n");
const [h, w, x, y] = input[0].split(" ").map(Number);
const arr = [];
for (let i = 1; i < input.length - 1; i++) {
  arr.push(input[i].split(" ").map(Number));
}
const a = Array.from(Array(h), () => new Array(w).fill(0));

for (let i = 0; i < h; i++) {
  for (let j = 0; j < w; j++) {
    if (i < x || j < y) {
      a[i][j] = arr[i][j];
      continue;
    }
    a[i][j] = arr[i][j] - a[i - x][j - y];
  }
}

const result = a.reduce((result, row) => {
  return result + row.join(" ") + "\n";
}, "");
console.log(result);

File: Implementation/python/16967.py
h,w,x,y = map(int,input().split())

arr = [list(map(int,input().split())) for _ in range(h+x)]
a = [[0 for _ in range(w)] for _ in range(h)]
for i in range(h):
  for j in range(w):
    if i-x < 0 or j-y < 0:
      a[i][j] = arr[i][j]
      continue
    a[i][j] = arr[i][j] - a[i-x][j-y]

for i in range(h):
  print(" ".join(map(str,a[i])))
